### PR TITLE
Add simple force-delay tests + clean-up

### PR DESCRIPTION
**Pull request by ana-pantilie**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

File: plutus-core/untyped-plutus-core/test/Transform/Simplify.hs [hunk 1]
```diff
@@ -227,6 +227,18 @@ multiApp = runQuote $ do
       app = mkIterAppNoAnn lam [mkConstant @Integer () 1, mkConstant @Integer () 2, mkConstant @Integer () 3]
   pure app
 
+forceDelayNoApps :: Term Name PLC.DefaultUni PLC.DefaultFun ()
```
Comment (michaelpj) [inline plutus-core/untyped-plutus-core/test/Transform/Simplify.hs:230]: btw you can write these with input files that have program syntax in them. That can be easier for simple programs like this! I'm not sure why this file has so many test cases defined by explicitly constructing ASTs...

Comment (ana-pantilie) [reply plutus-core/untyped-plutus-core/test/Transform/Simplify.hs:230]: Oh thanks for the tip! I can't seem to find any example, though, could you share one?

Comment (michaelpj) [reply plutus-core/untyped-plutus-core/test/Transform/Simplify.hs:230]: Hmm, weird, there just aren't any for UPLC? Not sure why. But look at e.g. `PlutusIR.Transform.Beta.Test`

Comment (zliu41) [reply plutus-core/untyped-plutus-core/test/Transform/Simplify.hs:230]: Yeah UPLC tests have always been done this way. The lack of types makes it easy to construct terms in Haskell. I find it at least as easy as writing UPLC by hand.

**Comment by ana-pantilie** [comment]
Forgot to mark this as ready for review.

I tested the simple `force-delay` cancellation algorithm we had before my previous PR on the tests I've added, and it works. If someone remembers some test for which it failed when we took a look at this, please let me know. My suspicion is that we just misread the complicated nesting of some force-application-lambda-delay when we were looking at those examples related to my previous PR.

I also addressed some of the comments I got on my PRs after they were merged.

**Review by zliu41 — APPROVED** [review]

**Pull request merged**